lib/fastlane/plugin/firebase_app_distribution/helper/firebase_app_distribution_auth_client.rb

```diff
@@ -0,0 +1,62 @@
+require 'fastlane_core/ui/ui'
+module Fastlane
+  UI = FastlaneCore::UI unless Fastlane.const_defined?("UI")
+  module Auth
+    module FirebaseAppDistributionAuthClient
+      TOKEN_CREDENTIAL_URI = "https://oauth2.googleapis.com/token"
+
+      def fetch_auth_token(param)
+        @google_service_path = param
+        auth_method = fetch_auth_method
+        case auth_method
+        when "none"
+          UI.crash!(ErrorMessage::MISSING_CREDENTIALS)
+        when "service_account"
+          service_account
+        when "firebase_token"
+          firebase_token
+        end
+      end
+
+      private
+
+      def fetch_auth_method
+        if !@google_service_path.nil? && !@google_service_path.empty?
+          "service_account"
+        elsif ENV["GOOGLE_APPLICATION_CREDENTIALS"]
+          @google_service_path = ENV["GOOGLE_APPLICATION_CREDENTIALS"]
+          "service_account"
+        elsif ENV["FIREBASE_TOKEN"]
+          "firebase_token"
+        else
+          "none"
+        end
+      end
+
+      def firebase_token
+        begin
+          client = Signet::OAuth2::Client.new(
+            token_credential_uri: TOKEN_CREDENTIAL_URI,
+            client_id: Fastlane::Actions::FirebaseAppDistributionLoginAction::CLIENT_ID,
+            client_secret: Fastlane::Actions::FirebaseAppDistributionLoginAction::CLIENT_SECRET,
+            refresh_token: ENV["FIREBASE_TOKEN"]
+          )
+        rescue Signet::AuthorizationError
+          UI.crash!(ErrorMessage::REFRESH_TOKEN_ERROR)
+        end
+        client.fetch_access_token!
+        client.access_token
```

Comment:
Unfortunately not. That method only tells the client to get the access token it doesn't actually return the access token in some form. It seems to actually fetch the access token and set it so that the next line is the line that actually retrieves the token.

Reply:
Ah, so `Signet::OAuth2::Client` and `Google::Auth::ServiceAccountCredentials` have similar but not identical interfaces I guess.... That's unfortunate.

Sounds good, thanks.